### PR TITLE
fix: move streamlit demo into conversant

### LINKED ISSUE #48: demo_utils.launch_streamlit unable to work on local install of `conversant`
The files of `conversant` that are installed by pip are only from `sandbox-conversant-lib/conversant`.

`demo_utils.launch_streamlit` requires `app/streamlit_example.py` to be available.

Code to replicate:
`pip install conversant`
```python
co = cohere.Client("API KEY")
shakespeare_config = {
    "preamble": "Below is a conversation between Shakespeare and a Literature Student.",
    "example_separator": "<CONVERSATION>\n",
    "headers": {
        "user": "Literature Student",
        "bot": "William Shakespeare",
    },
    "examples": [
        [
            {
                "user": "Who are you?",
                "bot": "Mine own nameth is Shakespeare, and I speaketh in riddles.",
            },
        ]
    ],
}
shakespeare_bot = PromptChatbot(
    client=co,
    prompt=ChatPrompt.from_dict(shakespeare_config),
    persona_name="william-shakespeare",
)
demo_utils.launch_streamlit(shakespeare_bot)
```

Error:
```
Usage: streamlit run [OPTIONS] TARGET [ARGS]...
Try 'streamlit run --help' for help.

Error: Invalid value: File does not exist: /app/streamlit_example.py
```

One possible solution is to move `app/` into `conversant` (with a rename), such that `streamlit_example.py` is available in the pip installed version of `conversant` as `conversant/demo/streamlit_example.py`.

### PR DESCRIPTION
**Pull request by yichern**
### What this PR does
- This packages `streamlit_example.py` into `conversant` so that it is installed alongside everything else by pip
- Allows `demo_utils.launch_streamlit` to launch a demo from a chatbot using `conversant/demo/streamlit_example.py`
- Clarify that an API key can be set as an environment variable or in `.streamlit/secrets.toml`
- Run `streamlit_example.py` from the absolute import path of `conversant`
- Set `PERSONA_MODEL_DIRECTORY` to be relative to the absolute import path of `conversant`

Note that the API key needs to be set as an environment variable before calling `demo_utils.launch_streamlit`

Closes #48 

### How it was tested
```bash
cd ~/mydir/sandbox-conversant-lib/
cd ..
pwd # prints ~/mydir
python ~/mydir/sandbox-conversant-lib/script.py
```
script.py is
```python
import cohere
import conversant
from conversant import PromptChatbot
from conversant.prompts import ChatPrompt

from conversant.utils import demo_utils

co = cohere.Client("API_KEY")
shakespeare_config = {
    "preamble": "Below is a conversation between Shakespeare and a Literature Student.",
    "example_separator": "<CONVERSATION>\n",
    "headers": {
        "user": "Literature Student",
        "bot": "William Shakespeare",
    },
    "examples": [
        [
            {
                "user": "Who are you?",
                "bot": "Mine own nameth is Shakespeare, and I speaketh in riddles.",
            },
        ]
    ],
}
shakespeare_bot = PromptChatbot(
    client=co,
    prompt=ChatPrompt.from_dict(shakespeare_config),
    persona_name="william-shakespeare",
)
os.environ["COHERE_API_KEY"] = "API_KEY"
demo_utils.launch_streamlit(shakespeare_bot)
```

Able to speak to `shakespeare_bot` on launched streamlit demo after running this script.

### PR checklist

- [x] No API keys or other secrets committed to source?
- [x] New functionality is added alongside appropriate tests?
- [x] All source files have the following header?

```
# Copyright (c) {YEAR} Cohere Inc. and its affiliates.
#
# Licensed under the MIT License (the "License");
# you may not use this file except in compliance with the License.
#
# You may obtain a copy of the License in the LICENSE file at the top
# level of this repository.
```

### CODE REVIEW

File: conversant/demo/streamlit_example.py [hunk 1]
```diff
@@ -32,7 +33,7 @@ def initialize_chatbot() -> None:
     """Initializes the chatbot from a selected persona and saves the session state."""
     if st.session_state.persona.startswith("from launch_demo") and len(sys.argv) > 1:
         st.session_state.bot = demo_utils.decode_chatbot(
-            sys.argv[1], client=cohere.Client(st.secrets.COHERE_API_KEY)
+            sys.argv[1], client=cohere.Client(os.environ.get("COHERE_API_KEY"))
```
Comment (eddotman) [inline conversant/demo/streamlit_example.py:36]: ohhh right because streamlit secrets are also injected into the env right? Nice!

**Review by eddotman — APPROVED** [review]
Ok this looks great! Also very cool way of testing the solution; thanks for documenting that so thoroughly.

File: conversant/demo/streamlit_example.py (hunk 1 above)
Comment (yichern) [reply conversant/demo/streamlit_example.py:36]: Yup! Over here: https://github.com/streamlit/streamlit/blob/e1ef414ea4893e8870e8416dceaabd11173ffd00/lib/streamlit/runtime/secrets.py#L197

**Comment by yichern** [comment]
This PR will break cohere-ai-sandbox-conversant-lib-appstreamlit-example-hgyk4x.streamlit.app, and we'll have to redeploy it to point to `conversant/demo/streamlit_example.py`

**Pull request merged**